### ISSUE TITLE
JSON deserialization error causes consumer thread to die?

### ISSUE THREAD

**Issue by s-porpoise** [spring-cloud/spring-cloud-stream-binder-aws-kinesis#40]
Hi,

Firstly, thanks for this binder it's amazing how smoothly it's been to set up and get working considering it's still very new.

One issue I'm running into if I try to receive JSON objects is that parsing errors result in a runtime exception that causes the consumer thread to die. I'm using spring.cloud.stream.bindings.input.consumer.concurrency=5, and can see five consumer threads logging every second, but after the error, the failing one never runs again. In my use case this means that the shard being read by that consumer will never be read again.

Example stacktrace:

```
2018-03-01 13:58:38.713 ERROR 5968 --- [-kinesis-consumer-1] o.s.integration.handler.LoggingHandler   : org.springframework.messaging.converter.MessageConversionException: Could not read JSON: Unexpected character ('r' (code 114)): was expecting double-quote to start field name
 at [Source: (byte[])"{ref: "XYZ123"}"; line: 1, column: 3]; nested exception is com.fasterxml.jackson.core.JsonParseException: Unexpected character ('r' (code 114)): was expecting double-quote to start field name
 at [Source: (byte[])"{reloc: "XYZ123"}"; line: 1, column: 3], failedMessage=GenericMessage [payload=byte[17], headers={aws_shard=shardId-000000000003, id=03d0e8c4-e9ee-d1a2-8188-08c13baadacf, contentType=application/json, aws_receivedStream=TestStream, aws_receivedPartitionKey=1, aws_receivedSequenceNumber=49582162487825732185088604198810591853496823934477140018, timestamp=1519876718698}]
	at org.springframework.messaging.converter.MappingJackson2MessageConverter.convertFromInternal(MappingJackson2MessageConverter.java:235)
	at org.springframework.cloud.stream.converter.ApplicationJsonMessageMarshallingConverter.convertFromInternal(ApplicationJsonMessageMarshallingConverter.java:86)
	at org.springframework.messaging.converter.AbstractMessageConverter.fromMessage(AbstractMessageConverter.java:181)
	at org.springframework.messaging.converter.CompositeMessageConverter.fromMessage(CompositeMessageConverter.java:70)
	at org.springframework.messaging.handler.annotation.support.PayloadArgumentResolver.resolveArgument(PayloadArgumentResolver.java:137)
	at org.springframework.messaging.handler.invocation.HandlerMethodArgumentResolverComposite.resolveArgument(HandlerMethodArgumentResolverComposite.java:116)
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.getMethodArgumentValues(InvocableHandlerMethod.java:137)
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:109)
	at org.springframework.cloud.stream.binding.StreamListenerMessageHandler.handleRequestMessage(StreamListenerMessageHandler.java:55)
	at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:109)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:141)
	at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:116)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:132)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:105)
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:73)
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:438)
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:388)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:181)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:160)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:108)
	at org.springframework.integration.endpoint.MessageProducerSupport.sendMessage(MessageProducerSupport.java:197)
	at org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter.access$3600(KinesisMessageDrivenChannelAdapter.java:86)
	at org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter$ShardConsumer.performSend(KinesisMessageDrivenChannelAdapter.java:957)
	at org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter$ShardConsumer.processRecords(KinesisMessageDrivenChannelAdapter.java:928)
	at org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter$ShardConsumer.lambda$processTask$1(KinesisMessageDrivenChannelAdapter.java:835)
	at org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter$ConsumerInvoker.run(KinesisMessageDrivenChannelAdapter.java:1018)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

I will probably work around this by receiving the message as a string and doing the JSON conversion manually, but it would be nice if the direct translation worked and handled errors without killing the consumer.

So basically I will be changing my consumer from something like

```
    @StreamListener(Sink.INPUT)
    public void handle(final Reservation reservation) {
```

To

```
    @StreamListener(Sink.INPUT)
    public void handle(final String reservation) {
        // manual deserialization with error handling
```
